bridger-cli/src/utils/transactionHandler.ts

```diff
@@ -0,0 +1,159 @@
+import { getBridgeConfig } from "../consts/bridgeRoutes";
+import { ClaimStruct } from "./claim";
+
+interface PendingTransactions {
+  claim: string | null;
+  verifySnapshot: string | null;
+  withdrawClaimDeposit: string | null;
+  startVerification: string | null;
+}
+
+/**
+ * Handles transactions for a given veaOutbox and epoch.
+ *
+ * @param chainId - The chainId of veaOutbox chain
+ * @param epoch - The epoch number for which the transactions are being handled
+ * @param veaOutbox - The veaOutbox instance to use for sending transactions
+ * @param claim - The claim object for the epoch
+ * @returns An instance of the TransactionHandler class
+ *
+ * @example
+ * const txHandler = new TransactionHandler(11155111, 240752, veaOutbox, claim);
+ * txHandler.sendTransaction(txData);
+ */
+
+export class TransactionHandler {
+  public epoch: number;
+  public veaOutbox: any;
+  public chainId: number;
+  public claim: ClaimStruct | null;
+  public getBridgeConfig: typeof getBridgeConfig;
+  public requiredConfirmations: number = 12;
+
+  public pendingTransactions: PendingTransactions = {
+    claim: null,
+    verifySnapshot: null,
+    withdrawClaimDeposit: null,
+    startVerification: null,
+  };
+
+  constructor(
+    chainId: number,
+    epoch: number,
+    veaOutbox: any,
+    claim?: ClaimStruct,
+    fetchBridgeConfig: typeof getBridgeConfig = getBridgeConfig
+  ) {
+    this.epoch = epoch;
+    this.veaOutbox = veaOutbox;
+    this.chainId = chainId;
+    this.claim = claim;
+    this.getBridgeConfig = fetchBridgeConfig;
+  }
+
+  public async checkTransactionPendingStatus(trnxHash: string | null): Promise<boolean> {
+    if (trnxHash == null) {
+      return false;
+    }
+
+    const receipt = await this.veaOutbox.provider.getTransactionReceipt(trnxHash);
+
+    if (!receipt) {
+      console.log(`Transaction ${trnxHash} is pending`);
+      return true;
+    }
+
+    const currentBlock = await this.veaOutbox.provider.getBlock("latest");
+    const confirmations = currentBlock.number - receipt.blockNumber;
+
+    if (confirmations >= this.requiredConfirmations) {
+      console.log(`Transaction ${trnxHash} is final with ${confirmations} confirmations`);
+      return false;
+    } else {
```

Comment:
Else not needed, as the return will already prevent any other branch. You can use the same guards pattern you have been flawlessly using in the rest of the function.

```ts
if (confirmations >= this.requiredConfirmations) {
  console.log(`Transaction ${trnxHash} is final with ${confirmations} confirmations`);
  return false;
}

console.log(`Transaction ${trnxHash} is not final yet.`);
return true;
```